core/src/upgrade.rs

```diff
@@ -134,6 +135,26 @@ impl<T: AsRef<[u8]>> ProtocolName for T {
     }
 }
 
+pub struct DisplayProtocolName<N>(N);
+impl<N: ProtocolName> Display for DisplayProtocolName<N> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        for byte in self.0.protocol_name() {
+            if (b' '..=b'~').contains(byte) {
+                f.write_char(char::from(*byte))?;
+            } else {
+                write!(f, "<{:02X}>", byte)?;
+            }
+        }
+        Ok(())
+    }
+}
```

Comment:
```suggestion
impl<N: ProtocolName> fmt::Display for DisplayProtocolName<N> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        for byte in self.0.protocol_name() {
            if (b' '..=b'~').contains(byte) {
                f.write_char(char::from(*byte))?;
            } else {
                write!(f, "<{:02X}>", byte)?;
            }
        }
        Ok(())
    }
}
```

I typically prefer this way of referring to the types because you only end up needing `use std::fmt;`

Reply:
okay, will do!